tools/ipapy2/ipapy2_MS1_annotation.xml

```diff
@@ -0,0 +1,105 @@
+<tool id="ipapy2_MS1_annotation" name="4. IPA MS1 annotation" version="@TOOL_VERSION@+galaxy0" python_template_version="3.5" profile="21.05">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${ipapy2_MS1_annotation_cli}'
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_MS1_annotation_cli">
+@init_logger@
+
+df = pd.read_csv('${MS1_table}')
+allAdds = pd.read_csv('${all_adducts}' )
+#if $ppmunk == "None"
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#end if
+#else
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#end if
+#end if
+annotations.to_csv('${output}', index=False)
+</configfile>
+</configfiles>
+
+    <inputs>
+        <param label="MS1 table" name="MS1_table" type="data" format="csv" help="pandas dataframe containing the MS1 data." />
+        <param label="all possible adducts table" name="all_adducts" type="data" format="csv" help="pandas dataframe containing the information on all the possible adducts given the database." />
```

Comment:
This is the database with the adducts added to it, right? Then I think the parameter should be renamed to database.